lib/inv_cg_quda.cpp

```diff
@@ -310,10 +346,30 @@ namespace quda {
     // compute initial residual
     double r2 = 0.0;
     if (param.use_init_guess == QUDA_USE_INIT_GUESS_YES) {
-      mat(r, x, y, tmp3);
-      r2 = blas::xmyNorm(b, r);
-      if (b2 == 0) b2 = r2;
-      blas::copy(y, x);
+
+      // Just replace any initial guess with a deflated RHS
```

Comment:
This is a bad idea: I've seen cases in MILC where sometimes CG is called to "polish" a solution (i.e., check and make sure it's a solution, and if it's not polish it up so it is, which is normally just a few iterations). In cases like that, completely nuking the existing initial guess is suboptimal. We should work out the logic to avoid this problem.

Reply:
That said, I don't know if there's an intersection of deflation and polishing solutions in practice in general, but still, this shouldn't be too difficult to work out.

Reply:
I agree, this should be fixed.  It would also come if we were at some point to combine deflation with multi-shift solvers, e.g., where we run deflated staggered multi shift in mixed precision and then want to refine using deflated CG.

Reply:
You form new residual equation, e.g., `r = b - Ax_0`, which is what you do normally with an initial guess.  Now the system you deflate is `A y = r`, and the total solution vector is given by `x = x0 + y`.